neuralpp/test/quick_tests/symoblic/simplication_test.py

```diff
@@ -0,0 +1,54 @@
+"""
+Tests of SymPy's simplication functions.
+https://docs.sympy.org/latest/tutorial/simplification.html
+"""
+from sympy import *
+
+
+def test_simplify() -> None:
+    """
+    One can just use sympy.simplify().
+    It's push-button but has 2 caveats:
+    1. "simplest" form is not well-defined. (showcased by last set of asserts)
+    2. it can be unnecessarily slow due to generality.
+    """
+    x = symbols('x')
+    assert simplify(sin(x)**2 + cos(x)**2) == 1
+    assert simplify((x**3 + x**2 - x - 1)/(x**2 + 2*x + 1)) == x - 1
+    assert simplify(gamma(x) / gamma(x - 2)) == (x - 2) * (x - 1)
+    # we may want to simplify (x**2 + 2*x + 1) into (x+1)**2 but that's not "simplest" recognized by simplify()
+    assert simplify(x**2 + 2*x + 1) != (x + 1)**2
+    # instead, we can call more specific simplification function.
+    assert factor(x**2 + 2*x + 1) == (x + 1)**2
+
+
+def test_expand() -> None:
+    """ The function expand() expands. """
+    x = symbols('x')
+    assert expand((x + 2)**2) == x**2 + 4*x + 4
+    assert expand((x + 2)*(x - 2)) == x**2 - 4
+
+
+def test_factor() -> None:
+    """ The function factor() turns a polynomial into an irreducible product of factors. """
```

Comment:
Does it work for arbitrary degrees? I don't know much about this but my understanding is that this is hard for higher degrees.

Reply:
Neither do I. From the tutorial:

> factor() uses a complete multivariate factorization algorithm over the rational numbers, which means that each of the factors returned by factor() is guaranteed to be irreducible.

So my guess is it works for arbitrary degrees though it might be slower for higher degrees.